src/providers/push/socketio/socketio.provider.ts

```diff
@@ -24,29 +24,24 @@ export class SocketIOProvider implements Provider<SocketNotification> {
     return {
       publish: async (message: SocketMessage) => {
         if (message.receiver.to.length === 0) {
-          throw new HttpErrors.BadRequest(
-            'Message receiver not found in request',
-          );
+          throw new HttpErrors.BadRequest('Message receiver not found');
         }
 
         /**
          * This method is responsible to send all the required data to socket server
          * The socket server needs to parse the data and send the message to intended
          * user.
+         *
+         * emitting a message to channel passed via config
          */
-        const publishes = message.receiver.to.map(receiver => {
-          const publishConfig = {
-            channel: receiver.id,
-            message: {
-              title: message.subject,
-              description: message.body,
-            },
-          };
 
-          return this.socketService.emit(JSON.stringify(publishConfig));
-        });
-
-        await Promise.all(publishes);
+        if (!this.socketConfig || !this.socketConfig.defaultChannelName) {
+          throw new HttpErrors.PreconditionFailed('Channel info is missing !');
+        }
+        await this.socketService.emit(
+          this.socketConfig.defaultChannelName,
```

Comment:
You should also allow channel name to be passed for each API, which will override default channel